### PR TITLE
Update add_dylink_section to work with wasm backend.

### PR DESCRIPTION
**Pull request by sbc100**
In this case its jobs is to add the dso dependencies since
that other fields are already present.

### CODE REVIEW

File: emscripten.py [hunk 1]
```diff
@@ -664,7 +664,8 @@ def align_memory(addr):
 
 
 def align_static_bump(metadata):
-  metadata['staticBump'] = align_memory(metadata['staticBump'])
+  if not shared.Settings.RELOCATABLE:
```
Comment (kripken) [inline emscripten.py:667]: why? (is this only for fastcomp?)

Comment (sbc100) [reply emscripten.py:667]: You are right this looks odd.   The reason is that the static bump gets written into the shared library as the static data size, which is separate from its alignment.

Let me see if I can refactor to make this look less odd.

**Comment by sbc100** [comment]
PTAL.. think this is good to go now.

File: emscripten.py (hunk 1 above)
Comment (sbc100) [reply emscripten.py]: This is not part of the change anymore

**Pull request merged**